airbyte-oauth/src/main/java/io/airbyte/oauth/flows/SurveymonkeyOAuthFlow.java

```diff
@@ -36,14 +47,29 @@ public SurveymonkeyOAuthFlow(final ConfigRepository configRepository, final Http
     super(configRepository, httpClient, stateSupplier);
   }
 
+  protected String getBaseURLByOrigin(final JsonNode inputOAuthConfiguration) throws Error {
+    final String origin = getConfigValueUnsafe(inputOAuthConfiguration, "origin");
+    if (EUROPE.equals(origin)) {
+      return API_ACCESS_URL_EU + AUTHORIZE_URL;
+    } else if (CANADA.equals(origin)) {
+      return API_ACCESS_URL_CA + AUTHORIZE_URL;
+    } else if (USA.equals(origin)) {
+      return API_ACCESS_URL_USA + AUTHORIZE_URL;
+    } else {
+      throw new Error("Unkown Origin");
```

Comment:
the error message should include the unknown origin to help the user troubleshoot the issue

Reply:
done